extras/userdoc/generator/modules/writers.py

```diff
@@ -0,0 +1,420 @@
+# -*- coding: utf-8 -*-
+#
+# writers.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+NEST help writer
+================
+
+Collect all the data and write help files.
+"""
+
+import os
+import re
+import textwrap
+from helpers import cut_it
+
+if os.path.isdir('../cmds'):
+    pass
+else:
+    os.mkdir('../cmds')
+if os.path.isdir('../cmds/sli'):
+    pass
+else:
+    os.mkdir('../cmds/sli')
+if os.path.isdir('../cmds/cc'):
+    pass
+else:
+    os.mkdir('../cmds/cc')
+
+
+def write_help_html(doc_dic, file, sli_command_list, keywords):
+    """
+    Write html.
+
+    Write html for integration in NEST Help-System
+    """
+    name = ""
+    header_ref = '''<!DOCTYPE html>
+    <html>
+        <head>
+    '''
+    header_style = '''
+    <style type="text/css">
+      body {
+        padding: 0;
+        margin: 0;
+      }
+      a {
+        color: #339;
+        text-decoration: none;
+      }
+      a:visited {
+        color: #339;
+        text-decoration: none;
+      }
+      a:hover {
+        text-decoration: underline;
+      }
+      h1 {
+        padding: 15px 0 0 15px;
+      }
+      p {
+        padding-left: 15px;
+      }
+      table.headerfooter {
+        margin: 20px 0 20px 0;
+        background-color: #eee;
+        width: 100%;
+        height: 30px;
+        border-top: 2px solid #ccc;
+        border-bottom: 2px solid #ccc;
+        text-align: center;
+      }
+      div.wrap {
+        padding-left: 15px;
+      }
+    </style>
+    '''
+
+    footer = '''
+        </body>
+    </html>'''
+
+    linkline = '''
+    <table class="headerfooter">
+      <tbody><tr>
+        <td width="30%" align="center"><a href="../../index.html">
+            NEST HelpDesk</a></td>
+        <td width="30%" align="center"><a href="../helpindex.html">
+            Command Index</a></td>
+        <td width="30%" align="center"><a href="../../quickref.html">
+            NEST Quick Reference</a></td>
+      </tr>
+    </tbody></table>
+    '''
+
+    copyright = '''
+    <p style="text-align:center">
+        &copy; 2016 <a href="http://www.nest-initiative.org">
```

Comment:
Since we copyright all code per 2004, I'd suggest to do so for the documentation, too.